raiden/tests/integration/transfer/test_refundtransfer.py

```diff
@@ -182,3 +177,140 @@ def test_refund_transfer(raiden_chain, token_addresses, deposit, network_wait):
+    )
+
+    # make a transfer to test the path app0 -> app1 -> app2 -> app3
+    identifier_path = 1
+    amount_path = 1
+    mediated_transfer(
+        app0,
+        app3,
+        token_network_identifier,
+        amount_path,
+        identifier_path,
+        timeout=network_wait,
+    )
+
+    # drain the channel app2 -> app3
+    identifier_drain = 2
+    amount_drain = deposit * 8 // 10
+    direct_transfer(
+        app2,
+        app3,
+        token_network_identifier,
+        amount_drain,
+        identifier_drain,
+        timeout=network_wait,
+    )
+
+    # wait for the nodes to sync
+    gevent.sleep(0.2)
+
+    assert_synched_channel_state(
+        token_network_identifier,
+        app0, deposit - amount_path, [],
+        app1, deposit + amount_path, [],
+    )
+    assert_synched_channel_state(
+        token_network_identifier,
+        app1, deposit - amount_path, [],
+        app2, deposit + amount_path, [],
+    )
+    assert_synched_channel_state(
+        token_network_identifier,
+        app2, deposit - amount_path - amount_drain, [],
+        app3, deposit + amount_path + amount_drain, [],
+    )
+
+    # app0 -> app1 -> app2 > app3 is the only available path, but the channel
+    # app2 -> app3 doesn't have capacity, so a refund will be sent on
+    # app2 -> app1 -> app0
+    identifier_refund = 3
+    amount_refund = 50
+    async_result = app0.raiden.mediated_transfer_async(
+        token_network_identifier,
+        amount_refund,
+        app3.raiden.address,
+        identifier_refund,
+    )
+    assert async_result.wait() is False, 'there is no path with capacity, the transfer must fail'
+
+    gevent.sleep(0.2)
+
+    # Lock structures with the correct amount
+
+    send_locked1 = raiden_events_must_contain_entry(
+        app0.raiden,
+        SendLockedTransfer,
+        {'transfer': {'lock': {'amount': amount_refund}}},
+    )
+    assert send_locked1
+
+    send_refund1 = raiden_events_must_contain_entry(app1.raiden, SendRefundTransfer, {})
+    assert send_refund1
+
+    lock1 = send_locked1.transfer.lock
+    refund_lock1 = send_refund1.lock
+    assert lock1.amount == refund_lock1.amount
+
+    send_locked2 = raiden_events_must_contain_entry(
+        app1.raiden,
+        SendLockedTransfer,
+        {'transfer': {'lock': {'amount': amount_refund}}},
+    )
+    assert send_locked2
+
+    send_refund2 = raiden_events_must_contain_entry(app2.raiden, SendRefundTransfer, {})
+    assert send_refund2
+
+    lock2 = send_locked2.transfer.lock
+    refund_lock2 = send_refund2.lock
+    assert lock2.amount == refund_lock2.amount
+    assert lock2.secrethash
+    assert lock2.expiration
+
+    # channels have the amount locked because of the refund message
+    assert_synched_channel_state(
+        token_network_identifier,
+        app0, deposit - amount_path, [lockstate_from_lock(lock1)],
+        app1, deposit + amount_path, [lockstate_from_lock(refund_lock1)],
+    )
+    # TODO: This fails.
```

Comment:
this doesnt fail anymore, correct?

Reply:
correct, the entire comment will be removed